Assets/Scripts/Flock.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Flock : MonoBehaviour
+{
+    [SerializeField] private FlockAgent agentPrefab;
+    List<FlockAgent> agents = new List<FlockAgent>();
+    [SerializeField] private FlockBehaviour behaviour;
+
+    [Range(1, 500)]
+    [SerializeField] private int startingCount = 250;
+    const float AgentDensity = 0.08f;
+
+    [Range(1f, 100f)]
+    [SerializeField] private float driveFactor = 10f;
+    [Range(1f, 100f)]
+    [SerializeField] private float maxSpeed = 5f;
+    [Range(1f, 10f)]
+    [SerializeField] private float neighbourRadius = 1.5f;
+    [Range(0f, 1f)]
+    [SerializeField] private float avoidanceRadiusMultiplier = 0.5f;
+
+    private float squareMaxSpeed;
+    private float squareNeighbourRadius;
+    public float SquareAvoidanceRadius { get; set; }
+
+    void Start()
+    {
+        this.squareMaxSpeed = this.maxSpeed * this.maxSpeed;
+        this.squareNeighbourRadius = this.neighbourRadius * this.neighbourRadius;
+        SquareAvoidanceRadius = this.squareNeighbourRadius * this.avoidanceRadiusMultiplier * this.avoidanceRadiusMultiplier;
+
+        for (int i = 0; i < this.startingCount; i++)
+        {
+            FlockAgent newAgent = Instantiate(
+                this.agentPrefab,
+                Random.insideUnitSphere * this.startingCount * AgentDensity,
+                Quaternion.Euler(Vector3.forward * Random.Range(0f, 360f)),
+                transform
+            );
+
+            newAgent.name = "Agent " + i;
+            newAgent.AgentFlock = this;
+            agents.Add(newAgent);
+        }
+    }
+
+    void Update()
+    {
+        foreach (FlockAgent agent in agents)
+        {
+            agent.GetNearbyObjects(neighbourRadius);
+
+            Vector3 velocity = behaviour.CalculateMove(agent, agent.Context, this);
```

Comment:
Question: afaik Boids uses an accumulation of rules that will together impact the current velocity of an agent, but here I'm seeing that you only calculate the movement based on one behaviour. Where do the various velocities calculated from each rule get added together?

Reply:
Combined move calculations are done in the `CompositeBehavior.cs` script